### PR TITLE
BACK-22: Set up database

### PR DESCRIPTION
**Pull request by DogukanGun**
## Description of Issue
Database setup

## How I implemented
I created a general class to use all database operations

### CODE REVIEW

File: .env
Comment (cnk1dev) [inline .env:1]: Please do not push environment variables to the remote repository. Add .env to .gitignore, and either undo your commit where you added the secrets and hard push (if it works), or create a new branch without them and cherry pick your changes from this branch (without the supabase secret).

File: README.md [hunk 1]
```diff
@@ -16,3 +16,5 @@ npm run dev
 `npm run prepare` installs the precommit hook, that will run Eslint and Prettier before you commit anything.
 
 `npm run dev` check [Graphiql](http://localhost:3000/graphiql) with your browser to test the GraphQL server.
+
+`npm install --save @supabase/supabase-js` installs necessary packages for Supabase database
```
Comment (cnk1dev) [inline README.md:20]: We don't need to explain how to install specific packages in the Readme, they should be installed automatically using npm install if they were added to package.json

File: src/database/database.manager.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    this.supabase = supabase
+  }
+
+  async fetchData(tableName: string): Promise<{ [p: string]: any }[] | null> {
+    const { data, error } = await this.supabase.from(tableName).select()
+    console.error(error)
+    return data
+  }
+
+  async fetchDataWWhere(
```
Comment (cnk1dev) [inline src/database/database.manager.ts:20]: typo

File: src/database/database.manager.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    tableName: string,
+    filter: SupabaseFilter
+  ): Promise<{ [p: string]: any }[] | null> {
```
Comment (cnk1dev) [inline src/database/database.manager.ts:23]: Please type correctly or suppress warning if there is no type for this in the supabase package

File: src/database/database.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { createClient } from '@supabase/supabase-js'
+import { SupabaseClient } from '@supabase/supabase-js/src'
+import { GenericSchema } from '@supabase/supabase-js/src/lib/types'
+
+class Database {
+  private static supabaseUrl = 'https://yipnhkmklmbuxjwzpipg.supabase.co'
+  private static key: string =
+    process.env.SUPABASE_ANON_KEY != null ? process.env.SUPABASE_ANON_KEY : ''
+  supabaseKey =
```
Comment (cnk1dev) [inline src/database/database.ts:9]: Remove key from code and add as environment variable (see comment above)

File: src/database/database.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6InlpcG5oa21rbG1idXhqd3pwaXBnIiwicm9sZSI6ImFub24iLCJpYXQiOjE2ODM0OTAwMjYsImV4cCI6MTk5OTA2NjAyNn0.5H4WqY1uUq9IiCGrPvM96NSrTLO6pPUUS1gMb4Ubxds'
+  // eslint-disable-next-line @typescript-eslint/no-unsafe-call,@typescript-eslint/no-unsafe-assignment,@typescript-eslint/ban-ts-comment
```
Comment (cnk1dev) [inline src/database/database.ts:11]: If you have to disable 10 eslint rules, something is not right, please take a look at this again

**Review by cnk1dev — CHANGES_REQUESTED** [review]

File: src/database/database.ts (hunk 4 above)
Comment (pczern) [reply src/database/database.ts:9]: Generate new key, this one is already on GitHub public

**Review by pczern — CHANGES_REQUESTED** [review]
Please explain why not use GraphQL Mutations to insert data in the tables

**Pull request closed without merging**